dev/make-distribution.sh

```diff
@@ -241,6 +241,10 @@ fi
 
 # Make R package - this is used for both CRAN release and packing R layout into distribution
 if [ "$MAKE_R" == "true" ]; then
+  echo "Make sure Spark jars folder contains all hadoop dependencies"
+  SPARK_JARS_DIR="$SPARK_HOME/assembly/target/scala-${SCALA_VERSION}/jars"
+  cp -n ${SPARK_HOME}/R/rjarsdep/target/scala-${SCALA_VERSION}/jars/* $SPARK_JARS_DIR
```

Comment:
As Felix said, you're undoing what the hadoop-provided does. So what's the point in enabling it in the first place?

If your goal is really to build with R and without hadoop-provided, you could play with `mvn dependency:build-classpath` and `SPARK_DIST_CLASSPATH`.